rosetta-lang/src/main/java/com/regnosys/rosetta/validation/RosettaSimpleValidator.xtend

```diff
@@ -1404,8 +1408,8 @@ class RosettaSimpleValidator extends AbstractDeclarativeRosettaValidator {
 	@Check
 	def checkReduceOperation(ReduceOperation o) {
 		checkNumberOfMandatoryNamedParameters(o.function, 2)
-		if (o.argument.RType != o.function.body.RType) {
-			error('''List reduce expression must evaluate to the same type as the input. Found types «o.argument.RType» and «o.function.body.RType».''', o, ROSETTA_FUNCTIONAL_OPERATION__FUNCTION)
+		if (o.argument.RMetaAnnotatedType.RType != o.function.body.RMetaAnnotatedType.RType) {
```

Comment:
I think these should include metadata. It should also be loosened to a subtype rather than exact equality.

Reply:
done